### DISCUSSION TITLE
How do I insert the `LaTeX` symbol in Typst?

### DISCUSSION THREAD

**Discussion by mawkler** [typst/typst#1732] [Q&A] [answered]
How do I insert the `LaTeX` glyph? I.e. the equivalent of doing `\latex` in LaTeX?

Asking for a friend 👀

**Comment by sitandr** [comment, marked as answer]
There is no command in standart library, but it is quite easy to make it yourself.

I took code from [typst-undergradmath](https://github.com/johanvx/typst-undergradmath) and slightly modified for a bit better similarity:

```
#let TeX = style(styles => {
  let e = measure(text(1em, "E"), styles)
  let T = "T"
  let E = text(1em, baseline: e.height / 4, "E")
  let X = "X"
  box(T + h(-0.1em) + E + h(-0.125em) + X)
})

#let LaTeX = style(styles => {
  let l = measure(text(1em, "L"), styles)
  let a = measure(text(0.7em, "A"), styles)
  let L = "L"
  let A = text(0.7em, baseline: a.height - l.height, "A")
  box(L + h(-0.3em) + A + h(-0.1em) + TeX)
})

#LaTeX
```

Produces result:

![image](https://github.com/typst/typst/assets/60141933/6e75a1fa-1767-4f68-8d34-64a49283e1a2)


Compared to

Logo:
![image](https://github.com/typst/typst/assets/60141933/0e4f400f-6640-4ef0-b193-4e58714beaa6)

`\LaTeX` in pdflatex (not sure, took from wiki):
![image](https://github.com/typst/typst/assets/60141933/9b2ffdad-11f0-4aaa-b685-c1cf05a9f5be)

`{{stylized LaTeX}}` in Wiki text itself:
![image](https://github.com/typst/typst/assets/60141933/ae1adb12-4476-40e6-8bfa-553eb7788bbf)


In KaTex there seem to be problems with `\LaTeX` command:
![image](https://github.com/typst/typst/assets/60141933/d5069594-c9f2-429c-8626-1216a0eaa4c3)

So I guess the above result is close enough. :)

**Reply by mawkler** [reply]
Nice! Thank you very much!

**Comment by adriandelgado** [comment]
This is my best try after tweaking @sitandr 's solution.
```js
#let TeX = style(styles => {
  set text(font: "New Computer Modern")
  let e = measure("E", styles)
  let T = "T"
  let E = text(1em, baseline: e.height * 0.31, "E")
  let X = "X"
  box(T + h(-0.15em) + E + h(-0.125em) + X)
})

#let LaTeX = style(styles => {
  set text(font: "New Computer Modern")
  let a-size = 0.66em
  let l = measure("L", styles)
  let a = measure(text(a-size, "A"), styles)
  let L = "L"
  let A = box(scale(x: 105%, text(a-size, baseline: a.height - l.height, "A")))
  box(L + h(-a.width * 0.67) + A + h(-a.width * 0.25) + TeX)
})
```
Produces result:
![image](https://github.com/typst/typst/assets/11708972/565c899d-4d5f-4a55-9e89-618280679c2a)

Compare to `\LaTeX`:
![image](https://github.com/typst/typst/assets/11708972/b5a3d0d1-8d7c-49d0-b388-672514a5a7df)

**Comment by swaits** [comment]
Here are versions which don't depend on `measure`, `style`, or `context`.

```typst
#let TeX = {
  set text(font: "New Computer Modern",)
  let t = "T"
  let e = text(baseline: 0.22em, "E")
  let x = "X"
  box(t + h(-0.14em) + e + h(-0.14em) + x)
}

#let LaTeX = {
  set text(font: "New Computer Modern")
  let l = "L"
  let a = text(baseline: -0.35em, size: 0.66em, "A")
  box(l + h(-0.32em) + a + h(-0.13em) + TeX)
}
```

**Comment by Moelf** [comment]
looks like someone made a package: https://typst.app/universe/package/metalogo

**Comment by swaits** [comment]
Thanks for pointing that out to me. I pushed a PR with some refactoring that way. :)

https://github.com/lonkaars/typst-metalogo/pull/3
